### PR TITLE
feat: dont open urls when CLI is running in CI

### PR DESCRIPTION
**Pull request by yuth**
Updated the CLI not to open browser window when running in CI mode

re #5973

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli-core/src/index.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export * from './isPackaged';
 export * from './cliConstants';
 export * from './deploymentSecretsHelper';
 export * from './deploymentState';
+export * as utils from './utils';
```
Comment (edwardfoyle) [inline packages/amplify-cli-core/src/index.ts:18]: I would prefer to export `openIfNotCI` directly. Having an exported member like `utils` will likely become a kitchen sink for all kinds of unrelated functions in the future

Also utils is a super overloaded word (as you saw when you had to rename some imports) so I think more specific export names are better

File: packages/amplify-cli-core/src/utils/open.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import ciInfo from 'ci-info';
+import { ChildProcess } from 'child_process';
+
+/**
+ * Helper function to Open stuff like URLs, files, executables. Cross-platform and opens only if its run in non-ci environmets
+ * This is wrapper for https://github.com/sindresorhus/open
+ * @param target The thing you want to open. Can be a URL, file, or executable.
+ * @param options
+ */
+export const openIfNotCI = (target: string, options: open.Options): Promise<ChildProcess | void> => {
```
Comment (edwardfoyle) [inline packages/amplify-cli-core/src/utils/open.ts:11]: a little cleaner to write:
```ts
export const openIfNotCI = ciInfo.isCI ? async () => {} : open
```

File: packages/amplify-cli-core/src/index.ts (hunk 1 above)
Comment (yuth) [reply packages/amplify-cli-core/src/index.ts:18]: I agree util is overused, but exposing this directly would also is ugly. I would have used `helper` but that too is over used. I am open for suggestion if we can namespace this to something meaningful.

Comment (edwardfoyle) [reply packages/amplify-cli-core/src/index.ts:18]: We could just name it 'open' (or maybe 'cliOpen' if we want to distinguish it from the npm package). We don't need to leak the fact that it's a noop in CI environments in the function name

File: packages/amplify-cli-core/src/utils/open.ts (hunk 2 above)
Comment (yuth) [reply packages/amplify-cli-core/src/utils/open.ts]: I think this is less readable than what we have and would stick to it

File: packages/amplify-cli-core/src/utils/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export * from './open';
```
Comment (edwardfoyle) [inline packages/amplify-cli-core/src/utils/index.ts:1]: nit can get rid of this file and change the export in the main index.ts to export directly from open.ts

**Review by edwardfoyle — APPROVED** [review]

File: packages/amplify-cli-core/src/utils/index.ts (hunk 3 above)
Comment (attilah) [reply packages/amplify-cli-core/src/utils/index.ts:1]: I'd keep the single level export for clarity as we probably not exporting everything in an upper level directory.

**Review by attilah — APPROVED** [review]

**Pull request merged**